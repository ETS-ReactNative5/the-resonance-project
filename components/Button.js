import React, { Component } from 'react';
import { StyleSheet, TouchableHighlight, Text } from 'react-native';
import Colors from '../constants/Colors';

export default class Button extends Component {
  render() {
    return (
      <TouchableHighlight
        style={[styles.button, this.props.style]}
        onPress={this.props.onPress}
        underlayColor={Colors.blue}
      >
        <Text style={[styles.text, this.props.textStyle]}>
          {this.props.title.toUpperCase()}
        </Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    width: '100%',
    backgroundColor: Colors.darkBlue,
    borderRadius: 6,
    paddingHorizontal: 24,
    paddingVertical: 16,
  },
  text: {
    color: Colors.white,
    textAlign: 'center',
    fontSize: 16,
  },
});
