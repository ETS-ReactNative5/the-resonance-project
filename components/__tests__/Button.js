import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import Button from '../Button';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the button', async () => {
    const tree = renderer.create(<Button title="Button" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
