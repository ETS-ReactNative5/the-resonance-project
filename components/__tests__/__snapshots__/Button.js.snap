// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#183f50",
        "borderRadius": 6,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 16,
          "textAlign": "center",
        },
        Object {},
      ]
    }
  >
    BUTTON
  </Text>
</View>
`;
